sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsModelFactory.cs

```diff
@@ -0,0 +1,340 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Azure.AI.TextAnalytics
+{
+    /// <summary>
+    /// Model factory that enables mocking for the Text Analytics library.
+    /// </summary>
+    public static class TextAnalyticsModelFactory
```

Comment:
So what is the benefit of having public factory method for internal constructors instead of just public constructors? Better plasticity for changes in new versions?

Reply:
So only classes that are intended to be constructed by customers in the mainline scenarios have public ctors. Model factories are for narrow use cases like mocking.